frontend/model/state.js

```diff
@@ -109,20 +109,36 @@ const mutations = {
   },
   setChatRoomReadUntil (state, { chatRoomId, messageHash, createdDate }) {
     const prevMentions = state.chatRoomUnread[chatRoomId].mentions
+    const prevOthers = state.chatRoomUnread[chatRoomId].others
+
     Vue.set(state.chatRoomUnread, chatRoomId, {
       readUntil: { messageHash, createdDate, deletedDate: null },
-      mentions: prevMentions.filter(m => new Date(m.createdDate).getTime() > new Date(createdDate).getTime())
+      mentions: prevMentions.filter(m => new Date(m.createdDate).getTime() > new Date(createdDate).getTime()),
+      others: prevOthers.filter(m => new Date(m.createdDate).getTime() > new Date(createdDate).getTime())
```

Comment:
@Silver-IT @SebinSong I'm not 100% sure about this....

```
 chatRoomUnread: {}, // [chatRoomId]: { readUntil: { messageHash, createdDate }, mentions: [{ messageHash, createdDate }], others: [{ messageHash, createdDate }] }
```

Does all of this look like a DRY violation? I see a lot of repetition...

Since I don't fully understand how the unread code works, I don't have a good suggestion on how to address this, but maybe one of you does have an idea on how we can make this more elegant and more DRY?

Reply:
For example, maybe it's possible to add `type` and consolidate all of these together into one array consisting of `{ messageHash, createdDate, type }` ?

```js
chatRoomUnread: [] // { messageHash, createdDate, type }
```

Reply:
Yeah. That's a good idea. We can define a type and use it.
```
{
  messageHash: string,
  createdDate: string,
  deletedDate?: string
}
```

Not sure to consolidate is good. If we do that, we should make `getters` and it could affect the performance, I think.

Reply:
@taoeffect I agree that can be a better idea. It's my first PR related to the chat-room area in the app and I think at this point, @Silver-IT has better knowledge with the full scope that the change to this `chatRoomUnread` field would impact. So, if you @Silver-IT agree with Greg's idea on this, would you want to take care of DRYing these as a separate issue once this PR is merged?

Reply:
Sure. I can handle it.